worker/peergrouper/worker_test.go

```diff
@@ -659,7 +701,7 @@ func (s *workerSuite) TestWorkerRetriesOnPublishError(c *gc.C) {
 			c.Logf("** publish call: count=%d", count)
 			publishCh <- apiServers
 			count++
-			if count <= 3 {
+			if count < succeedOnAttempt {
```

Comment:
given you changed the off-by-one, I take it this is now succeeding on the 3rd attempt, rather than after the 3rd attempt.
I don't know any reason why that's particularly good/bad though I'm curious the motivation to change it.

Reply:
This was done while addressing a bunch of intermittent failures that were introduced by having a new voyeur.Value for the controller config. In the end I diagnosed the issue as being a race for the first channel read in the working loop. If it was for a controller config change, then this pre-dated the loop run for topology change and the loop threw an error because of an unrecognised voting machine.

I left the change, as it is semantically clearer.